src/Driver/Driver.jl

```diff
@@ -107,32 +137,64 @@ function parse_commandline(custom_settings)
         help = "interval at which to show simulation updates"
         metavar = "<interval>"
         arg_type = String
-        default = "60secs"
+        default = if haskey(defaults, :show_updates)
+            defaults[:show_updates]
+        else
+            get(ENV, "CLIMATEMACHINE_SETTINGS_SHOW_UPDATES", Settings.show_updates)
+        end
```

Comment:
Maybe something like the following? A function might be more compact...
```
macro get_default(setting, defaults_dict, defaults_struct)
    quote
        let setting = $(esc(setting)),
            defaults_dict = $(esc(defaults_dict)),
            defaults_struct = $(esc(defaults_struct)),
            setting_sym = Symbol(setting),
            setting_upper = uppercase(setting),
            env_var = string("CLIMATEMACHINE_SETTINGS_", setting_upper)
        if haskey(defaults_dict, setting_sym)
            defaults_dict[setting_sym]
        else
            get(ENV, env_var, Base.getproperty(defaults_struct, setting_sym))
        end
        end
    end
end
```

Reply:
In general I loathe macros, I feel they are overused :) A function would be better. There is some custom processing / validation that need to happen for each setting so I don't know if it's worth it to consolidate, so the loc gain would be minimal at the moment.  maybe something to do later when there are more settings depending on how this evolves?

Reply:
Ah, I see the `parse(Int, ...)` and the `uppercase()...`, yeah. Messy, but no way out. :-/

Reply:
What about

```julia
function get_default(Settings, defaults, sym)
    default = if haskey(defaults, sym)
        defaults[sym]
    else
        get(
            ENV,
            "CLIMATEMACHINE_SETTINGS_"*upper(String(sym)),
            getproperty(Settings, sym),
        )
    end
end
```

Reply:
I think we will need to move to a parameter framework that just generalizes everything so you can support things like running with a config / env overloads/ cli generation / and run time configuration that would encapsulate a lot of these details.  Maybe something like this already exists in the Julia world that we can re-use.  I don't want to do that right now though, or if it's worth it until other details get sorted out.

Reply:
I guess the question is, how many custom cases do we need to account for? If it's fewer than 5, I think it's worth writing some helper scripts / using existing Julia tools to simplify things. But I'm fine tackling this in a later PR.

Reply:
> I think we will need to move to a parameter framework that just generalizes everything so you can support things like running with a config / env overloads/ cli generation / and run time configuration that would encapsulate a lot of these details. Maybe something like this already exists in the Julia world that we can re-use. I don't want to do that right now though, or if it's worth it until other details get sorted out.

Absolutely. 💯  See #947 for where we need to take this entire layer at some point.